util/util.go

```diff
@@ -644,3 +644,14 @@ func ManagerDelegatingClientFunc(cache cache.Cache, config *rest.Config, options
 		StatusClient: c,
 	}, nil
 }
+
+// FinalizerExists check if the given finalizer exists in the object o
+func FinalizerExists(o metav1.Object, finalizer string) bool {
+	f := o.GetFinalizers()
+	for _, e := range f {
+		if e == finalizer {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
I really wish we could leverage `controllerutil.ContainsFinalizer` here rather than having to add a helper for this here. @vincepri is it worth backporting https://github.com/kubernetes-sigs/controller-runtime/commit/b999e72d531f70cdf877468514b1bf11a8badfd4#diff-4c146d38c03cf55e321de9ad53af4452 to controller-runtime v0.5.x for that purpose?

Reply:
We can, I can cut a release today if you're able to backport it to `release-0.5`

Reply:
https://github.com/kubernetes-sigs/controller-runtime/pull/1000

Reply:
https://github.com/kubernetes-sigs/controller-runtime/releases/tag/v0.5.6